spec/services/school_data_importer_spec.rb

```diff
@@ -0,0 +1,132 @@
+require "rails_helper"
+require "csv"
+
+RSpec.describe SchoolDataImporter do
+  let(:school_data_importer) { SchoolDataImporter.new }
+
+  describe "#run" do
+    let(:date_string) { Time.zone.now.strftime("%Y%m%d") }
+    let(:file) { spy(:file) }
+    let(:gias_csv_url) { "http://ea-edubase-api-prod.azurewebsites.net/edubase/edubasealldata#{date_string}.csv" }
+    let(:csv_file_name) { "tmp/schools_data_#{date_string}.csv" }
+    let(:row) do
+      {
+        "URN" => "12345",
+        "EstablishmentName" => "Penistone Grammer School",
+        "Street" => "123 High Street",
+        "PhaseOfEducation (code)" => "4",
+        "LA (code)" => "40",
+        "LA (name)" => "Blackpool",
+        "EstablishmentTypeGroup (code)" => "10",
+        "TypeOfEstablishment (code)" => "22",
+      }
+    end
+    let!(:request) { stub_request(:get, gias_csv_url).to_return(body: "foo", status: 200) }
+    before do
+      allow(File).to receive(:open).with(csv_file_name, "wb").and_yield(file)
+      allow(File).to receive(:delete).with(csv_file_name)
+      allow(CSV).to receive(:foreach)
+        .with(csv_file_name, {headers: true, encoding: "windows-1251:utf-8"})
+        .and_yield(row)
+    end
+
+    it "downloads and saves the Get Information About Schools CSV file" do
+      school_data_importer.run
+
+      expect(request).to have_been_requested
+      expect(file).to have_received(:write).with("foo")
+    end
+
+    context "when the download is successful" do
+      it "imports each row as a school with associated Local Authority" do
+        school_data_importer.run
+
+        imported_school = School.first
+        expect(imported_school.urn).to eql(12345)
+        expect(imported_school.name).to eql("Penistone Grammer School")
+        expect(imported_school.street).to eql("123 High Street")
+        expect(imported_school.phase).to eql("secondary")
+        expect(imported_school.school_type_code).to eql(22)
+        expect(imported_school.school_type_group_code).to eql(10)
+        expect(imported_school.local_authority.code).to eql(40)
+        expect(imported_school.local_authority.name).to eql("Blackpool")
+      end
+
+      context "when the school already exists" do
```

Comment:
I think a similar spec for when the local authority exists would also be helpful in terms of coverage and confidence the code is doing what we want.